src/src/rx-serial/SerialTramp.cpp

```diff
@@ -42,8 +43,11 @@ void SerialTramp::sendQueuedData(uint32_t maxBytesToSend)
         _fifo.unlock();
         setTXMode();
         _outputPort->write(frame, frameSize);
-        setRXMode();
         bytesWritten += frameSize;
+        lastSendTime = millis();
+    }
+    if (_fifo.size() == 0 && millis() - lastSendTime > 100) {
```

Comment:
Rather than using a time based check you could check if the TX is idle
```cpp
        if (_fifo.size() == 0 && uart_ll_is_tx_idle(UART_LL_GET_HW(1)))
```

Reply:
Oh that's so much better thank you